### ISSUE TITLE
false positive "Schema of a JSON object has no properties defined" when schema has allOf

### ISSUE THREAD

**Issue by DavidBiesack** [42Crunch/vscode-openapi#143]
The audit flags as "Schema of a JSON object has no properties defined" a schema that uses `allOf` schema composition and each of the nested schemas are `type: object` with properties.

For example, this schema is flagged:

```
components:
  schemas:
    composed:
      title: Composed Schema
      description: A Thing
      type: object
      allOf:
        - $ref: '#/components/schemas/base'
        - type: object
          properties:
            a:
               type: string
              ...
    base:
      title: Base
      description: Base
      type: object
      properties:
        a:
           type: string
           ...
  
```

**Comment by ak1394** [comment]
Thanks for reporting, will investigate!

**Comment by ak1394** [comment]
Fixed in version 4.8.0
